rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp

```diff
@@ -147,7 +161,22 @@ namespace rsx
 			case CELL_GCM_TEXTURE_A4R4G4B4:
 			{
 				glPixelStorei(GL_UNPACK_ROW_LENGTH, pitch / 2);
-				glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, tex.width(), tex.height(), 0, GL_RGBA, GL_UNSIGNED_SHORT_4_4_4_4, pixels);
+				
+				if (is_swizzled)
+				{
+					u16 height = tex.height();
+					u16 width = tex.width();
+
+					std::unique_ptr<u16[]> swz_buf(new u16[width * height]);
```

Comment:
same, std::vector<u16> and std::unique_ptr<u16[]> are almost equivalent, they use RAII and embed an array.